src/main/java/fi/aalto/cs/apluscourses/intellij/model/APlusProject.java

```diff
@@ -163,7 +168,11 @@ public void addCourseFileEntry(@NotNull Module module) throws IOException {
       Iterable<String> moduleNames = modulesObject::keys;
       for (String moduleName : moduleNames) {
         String moduleId = modulesObject.getJSONObject(moduleName).getString("id");
-        modules.put(moduleName, moduleId);
+        ZonedDateTime downloadedAt = ZonedDateTime
+            .parse(modulesObject.getJSONObject(moduleName).getString("downloadedAt"));
+        IntelliJModuleMetadata intelliJModuleMetadata = new IntelliJModuleMetadata(moduleId,
+            downloadedAt);
+        modules.put(moduleName, intelliJModuleMetadata);
```

Comment:
Possibility to avoid getting the module object twice:
```
JSONObject moduleObject = modulesObject.getJSONObject(moduleName);
String moduleId = moduleObject.getString("id");
ZonedDateTime downloadedAt = ZonedDateTime.parse(moduleObject.getString("downloadedAt"));
```